internal/translation/privateendpoint/conversion.go

```diff
@@ -0,0 +1,475 @@
+/*
+Copyright 2024 MongoDB.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package privateendpoint
+
+import (
+	"strings"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/cmp"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/status"
+)
+
+const (
+	ProviderAWS   = "AWS"
+	ProviderAzure = "AZURE"
+	ProviderGCP   = "GCP"
+
+	StatusInitiating        = "INITIATING"
+	StatusPending           = "PENDING"
+	StatusPendingAcceptance = "PENDING_ACCEPTANCE"
+	StatusWaitingForUser    = "WAITING_FOR_USER"
+	StatusVerified          = "VERIFIED"
+	StatusFailed            = "FAILED"
+	StatusRejected          = "REJECTED"
+	StatusDeleting          = "DELETING"
+)
+
+type EndpointService interface {
+	ServiceID() string
+	EndpointInterfaces() EndpointInterfaces
+	Provider() string
+	Region() string
+	Status() string
+	ErrorMessage() string
+}
```

Comment:
Why the getters? Is it because these fields must be read only?

Reply:
The main motivation was to have reasonable interface representation.

Reply:
Ah I see, I guess some places use the interface rather than the underlying type directly.